build.wake

```diff
@@ -115,83 +139,112 @@ def setVersion release file =
 def tarball Unit =
     def releaseResult = buildAs Unit
     def timeResult = buildOn Unit
+
     require Pass release = releaseResult
     require Pass time = timeResult
+
     # Create debian + RedHat package files
-    require Pass changelog =
-        setVersion release "debian/changelog"
-    require Pass spec =
-        setVersion release "wake.spec"
+    require Pass changelog = setVersion release "debian/changelog"
+    require Pass spec = setVersion release "wake.spec"
+
     # Identify those sources files to include in the tarball
-    require Pass allSources =
-        sources "." `.*`
+    require Pass allSources = sources "." `.*`
+
     def srcs =
         allSources
-        | filter (! matches `(debian/|wake.spec).*` _.getPathName) # omit packaging files
-        | filter (! matches `(\.circleci/|\.github/).*|\.dockerignore` _.getPathName) # omit CI files
-        | filter (! matches `(.*/)*\.gitignore` _.getPathName) # omit git files
+        | filter (!matches `(debian/|wake.spec).*` _.getPathName) # omit packaging files
+        | filter (!matches `(\.circleci/|\.github/).*|\.dockerignore` _.getPathName) # omit CI files
+        | filter (!matches `(.*/)*\.gitignore` _.getPathName) # omit git files
```

Comment:
This surprises me. How do we handle double negation? Are we sure we add the needed space so that it doesn't turn into a new operator?

Also because `!` has a funny precedence in Wake the standard is to leave it with a space (which is why this was done. The AST here is `NOT (APP(APP(matches, `...`), DOT(_, getPathName))` but I think the no space version implies that its bound to the `matches` word which it isn't....really `!` is kind of awful in Wake

Reply:
Yeah, I assumed this would get a comment :)

One thought I had (probably for the static analyzer), we should force parenthesis when `!` is applied to something "not simple" (same definition of simple we use for `if`) instead of using spaces to hint at the idea here.

Reply:
I honestly just want to ban `!` in wake. It isn't needed as often and you can almost always replace it with a `| not` (assuming we have a `not` function)

Reply:
https://github.com/sifive/wake/pull/1232 That only applied to `!`, please lmk if there are other unary op that is should apply to (with code examples please)

Reply:
I think you need to do it for any unary operator whos child isn't a primitive expression.